### PR TITLE
Fix goal achieved message #2337

### LINKED ISSUE #2337: Goal achieved message is not getting display
## Issue Overview
See the Video for more information: http://youtu.be/1dVLsd4t9Fs?hd=1

No able to see the Goal is achieved message when the Donation from `Display Options` is set to `button`

image

image

image

## Expected Behavior
Show the message that goal is being achieved 

## Current Behavior
No message is getting display

## Steps to Reproduce (for bugs)
See the Video

## Todos
- [x] Tests
- [x] Documentation

## WordPress Environment
<details>
```
### WordPress Environment ###

Home URL: http://localgive.dev
Site URL: http://localgive.dev
WP Version: 4.8.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
Permalink Structure: /%postname%/
Show on Front: posts
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval, give_subscription

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: localgive.dev
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: Apache/2.4.10 (Debian)
PHP Version: 7.1.4
PHP Post Max Size: 1,000 MB
PHP Time Limit: 900
PHP Max Input Vars: 3000
PHP Max Upload Size: 1,000 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.1t - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: ❌ 5.5.55 - We recommend a minimum MySQL version of 5.6. See: WordPress Requirements
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 1.8.17
Upgraded From: 1.8.17
Test Mode: Enabled
Currency Code: GNF
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://localgive.dev/donation-confirmation/
Failure Page: http://localgive.dev/donation-failed/
Donation History Page: http://localgive.dev/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation, PayPal Standard
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Admin Email Notifications: Enabled
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Enabled
Session Name: PHPSESSID
Cookie Path: /
Save Path: 
Use Cookies: Enabled
Use Only Cookies: Enabled

### Active Give Add-ons ###

Give - Fee Recovery: ❌ Unlicensed – by WordImpress – 1.3
Give - Recurring Donations: ❌ Unlicensed – by WordImpress – 1.4

### Other Active Plugins ###

Query Monitor: by John Blackbourn – 2.17.0

### Inactive Plugins ###

Debug Bar Console: by koopersmith – 0.3
Debug Bar Cron: by Zack Tollman, Helen Hou-Sandi – 0.1.2
Debug Bar Extender: by Thorsten Ott, Automattic – 0.5
Developer: by Automattic – 1.2.6
Duplicate Post: by Enrico Battocchi – 3.2
Give - 2Checkout Gateway: by WordImpress – 1.0.2
Give - AmeriCloud Payments: by AmeriCloud Solutions, Inc. – 1.2.0
Give - Authorize.net Gateway: by WordImpress – 1.3
Give - CCAvenue Gateway: by WordImpress – 1.0
Give - Currency Switcher: by WordImpress – 1.0.0
Give - Email Reports: by WordImpress – 1.0.1
Give - Form Field Manager: by WordImpress – 1.2.4
Give - Gift Aid: by WordImpress – 1.0.0
Give - Manual Donations: by WordImpress – 1.2
Give - PayPal Pro Gateway: by WordImpress – 1.1.4
Give - PDF Receipts: by WordImpress – 2.0.7
Give - PDF Receipts: by WordImpress – 2.0.7
Give - Per Form Emails: by WordImpress – 1.0.2
Give - Razorpay: by WordImpress – 1.1.0
Give - Stripe Gateway: by WordImpress – 1.5
Give - Tributes: by WordImpress – 1.2
Give - WePay Gateway: by WordImpress – 1.3.1
Give - WP All Import Addon: by Ben Meredith – 1.0
Give - Zapier: by WordImpress – 1.1.2
Give Cutom Plugin: by  –
Give to ZBS CRM Connector: by https://zerobscrm.com – 1.0.1
Hookr: by @explodybits – 1.0.0 (beta)
Loco Translate: by Tim Whitlock – 2.0.16
Log Deprecated Notices: by Andrew Nacin – 0.4
Log Viewer: by Markus Fischbacher – 14.05.04
Redis Object Cache: by Till Krüss – 1.3.5
Rewrite Rules Inspector: by Daniel Bachhuber, Automattic – 1.2.1
rtMedia for WordPress, BuddyPress and bbPress: by rtCamp – 4.4.4
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com, DavidAnderson – 1.13.12
User Switching: by John Blackbourn – 1.3.0
WooCommerce: by Automattic – 3.2.3
WordPress Importer: by wordpressdotorg – 0.6.3
WP Downgrade | Specific Core Version: by Reisetiger – 1.1.4
WP Rollback: by WordImpress – 1.5
WP Super Cache - Clear all cache: by Apasionados.es – 1.4

### Active MU Plugins ###

Local by Flywheel Relative URL (for Live Links): by Flywheel – 1.0

### Theme ###

Name: Twenty Sixteen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

```
</details>

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
This PR will fix #2337 
http://recordit.co/WbDAbF324n

## How Has This Been Tested?
Manually Tested

## Screenshots (jpeg or gifs if applicable):
image

image

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by raftaar1191** [comment]
@DevinWalker @ravinderk 

Please review and merge the PR

**Comment by ravinderk** [comment]
@raftaar1191 Issue #2337 clearly says that it needs PHP unit test. So it is not ready to review yet.

image

**Comment by raftaar1191** [comment]
Ooo @ravinderk 

Adding that

**Comment by raftaar1191** [comment]
@ravinderk PHPUnit Add 

Please review and merge it

File: tests/unit-tests/tests-donate-form-class.php [hunk 1]
```diff
@@ -97,4 +97,38 @@ public function test_get_goal() {
...
+		$class = $simple_form->get_form_wrap_classes( array( 'display_style' => 'button' ) );
+		$this->assertContains('give-display-button', $class);
+		$this->assertContains('give-display-button-only', $class);
+
+		// Update the DB and  checking if get_form_wrap_classes is fetching the proper class from the give meta.
+		$payment_display = give_get_meta( $this->_simple_form->ID, '_give_payment_display', true );
+		give_update_meta( $this->_simple_form->ID, '_give_payment_display', 'modal' );
+		$class = $simple_form->get_form_wrap_classes( array() );
+		$this->assertContains('give-display-modal', $class);
+		$payment_display = give_update_meta( $this->_simple_form->ID, '_give_payment_display', $payment_display );
```
Comment (ravinderk) [inline tests/unit-tests/tests-donate-form-class.php]: @raftaar1191 What is this line for?

**Comment by ravinderk** [comment]
@DevinWalker One test failed on this PR which I think we can ignore.
For ref: https://travis-ci.org/WordImpress/Give/jobs/302828451

screen shot 2017-11-16 at 10 28 06 am

**Pull request merged**